### ISSUE TITLE
Issue with the Taylor diags

### ISSUE THREAD

**Issue by cecilehannay** [NCAR/ADF#184]
### ADF run type

Model vs. Model

### What happened?

I am having a few issues with the Taylor diagram plot:
- in model vs model: it only generates the Taylor diagram for the case but it doesn't add the baseline.
![Screen Shot 2022-06-15 at 4 56 43 PM](https://user-images.githubusercontent.com/9723220/173955796-98d29ab5-22a4-4bb1-bd2e-04420db5bac5.png)

- in model vs obs: the taylor diags doesn't run 
It fails at: ```data_loc = adfobj.get_basic_info("obs_climo_loc", required=True)```

### ADF Hash you are using

f3efbaa

### What machine were you running the ADF on?

CISL machine

### What python environment were you using?

NPL (CISL machines only)

**Comment by brianpm** [comment]
For model vs model, this is the correct behavior. The baseline acts as the reference, so correlation would be with itself (=1).

For obs, we will need to look into it. It might just be that we don't have observational datasets for the variables.

**Comment by nusbaume** [comment]
In terms of observations, it looks like the Taylor diagram code was written before the new obs framework was brought in, so part of it needs to be re-written to use the variable defaults file to find the relevant observations, similar to the other, more standard plotting scripts (e.g. `global_latlon_map.py`).

Beyond that, however, I agree that some of the variables currently may not have an observational datasets to compare against.

**Comment by cecilehannay** [comment]
Thanks for the clarification about the model vs model behavior. It makes sense. At first, I was expecting a plot like in the AMWG diags where the case and the baseline are compared to a milestone simulation (like cesm1 or cesm2). 
I guess I stared for too many years at the AMWG Taylor Diag.

**Comment by justin-richling** [comment]
Would it be worth exploring having the Taylor plots be skipped if all the required variables aren't present in the datasets? Currently if there are any missing necessary vars it kills the ADF, which seem less than ideal. Just a thought.

**Comment by cecilehannay** [comment]
I fully agree. Indeed, I had started this feature to implemented this feature in my branch. It is just a check that if variables are not there, it skips the Taylor diag.

**Comment by cecilehannay** [comment]
We could add an option to select either:
(1) plot the test case versus the baseline (current behavior)
(2) plot the test case and baseline versus a milestone, for instance cmip6 runs (similar to what was done in the amwg diags).  In that case, the milestone should be different whether it is a F case or a B case.

**Comment by nusbaume** [comment]
Just to add this issue, I think it would be ideal if the Taylor diagram used the regridded data instead of the climo files directly, as it would allow it to skip the relatively expensive interpolation steps, and allow it to work with more varied types of vertical grids (e.g. MPAS). 

However, while trying to implement this change in PR #186 I found that the vertical integration code does not work correctly due to the presence of NaNs at the surface (for areas with topography above 1000 mb).  This will be fixable once the ECMWF method for hybrid -> pressure level interpolation is implemented, as described in this GeoCAT issue:

https://github.com/NCAR/geocat-comp/issues/207

Until then I think we are stuck as-is.

**Comment by brianpm** [comment]
I think we at least addressed the original issue in commits 8dc6938, bf74b97 — now cleanly skips ([cam_taylor_diagram.py:40](vscode-webview://1083b8ipmmcolvv7nlt9p0pnvvvsqkljsihrs4o588lnqdtduh4o/scripts/plotting/cam_taylor_diagram.py#L40))

There might still be some stray threads here, but closing this issue for now since no one has complained about it for several years.
